### ISSUE TITLE
[java] OverrideBothEqualsAndHashCodeOnComparable on class with lombok.EqualsAndHashCode annotation

### ISSUE THREAD

**Issue by kdebski85** [pmd/pmd#6096]
**Affects PMD Version:** 
7.17.0

**Rule:**
https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_errorprone.html#overridebothequalsandhashcodeoncomparable

**Description:**

`OverrideBothEqualsAndHashCodeOnComparable` is reported when:

- Comparable is implemented
- lombok.EqualsAndHashCode annotation is used

**Code Sample demonstrating the issue:**

```java
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;

@EqualsAndHashCode
@RequiredArgsConstructor
public class RuleTest implements Comparable<RuleTest> {

    private final String x;

    @Override
    public int compareTo(RuleTest other) {
        return x.compareTo(other.x);
    }
}
```

**Expected outcome:**

PMD reports a violation at line 11, but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Gradle

**Comment by zman0900** [comment]
This should consider lombok's `@Data` and `@Value` too.
